payload_generator/bruteforce.py

```diff
@@ -0,0 +1,54 @@
+# Auxiliary variables/constants for payload generation.
+
+INITIAL_VALUE = 0; # set this when you need to continue an interrupted fuzzing session 
+count = INITIAL_VALUE;
+user = str('admin');
+passfile_path = 'C:\\Users\\user\\Documents\\wordlists\\passwords.txt';
+NUMBER_OF_PAYLOADS = sum(1 for line in open(passfile_path));
+passwd = list();
+for line in open(passfile_path):    # initializing passwords into list
+    passwd.append(line.rstrip());
+print('NUMBER_OF_PAYLOADS = ' + str(NUMBER_OF_PAYLOADS));
+print('len(passwd) = ' + str(len(passwd)));
+print('count = '+str(count));
+
+
+# The number of generated payloads, zero to indicate unknown number.
+# The number is used as a hint for progress calculations.
+def getNumberOfPayloads():
+    return NUMBER_OF_PAYLOADS;
+
+
+# Returns true if there are still payloads to generate, false otherwise.
+# Called before each call to next().
+def hasNext():
+    return (count < NUMBER_OF_PAYLOADS);
+
+
+# Returns the next generated payload.
+# This method is called while hasNext() returns true.
+def next():
+    global count;
+    print('next_count = ' + str(count));
+    payload = count;
+    print('payload = ' + str(payload));
+    count+=1;
+    print('incremented next_count = ' + str(count));
+    print(user+':'+passwd[payload]);
+    return user+':'+passwd[payload];
```

Comment:
This could return the pair username/password already Base64 encoded, for example:
```Python
import base64
...

return base64.b64encode(user+':'+passwd[payload]);
```
avoids the need to manually add the payload processor, which is error prone.